tests/test_provider_configuration.py

```diff
@@ -57,41 +60,154 @@ def test_should_not_fetch_provider_metadata_if_given(self):
+        assert provider_config._client_metadata['registration_client_uri'] == 'https://op.example.com/register/client1'
+        assert provider_config._client_metadata['registration_access_token'] == 'registration_access_token1'
+        assert provider_config._client_metadata['redirect_uris'] == ['https://client.example.com/redirect',
+                                                                     'https://client.example.com/logout']
+        assert provider_config._client_metadata[
+                   'post_logout_redirect_uris'] == ['https://client.example.com/logout']
 
-class TestOIDCData(object):
+    @responses.activate
+    def test_register_client_should_not_create_redirect_uris_key_if_already_passed(self):
+
+        registration_endpoint = self.PROVIDER_BASEURL + '/register'
+        client_registration_response = {
+            'client_id': 'client1',
+            'client_secret': 'secret1',
+            'client_name': 'Test Client',
+            'redirect_uris': ['https://client.example.com/redirect', 'https://client.example.com/logout',
+                              'https://client.example.com/swaggerui/oauth2-redirect.html'],
+            'registration_client_uri': 'https://op.example.com/register/client1',
+            'registration_access_token': 'registration_access_token1'
+        }
+        responses.add(responses.POST, registration_endpoint, json=client_registration_response)
+        provider_config = ProviderConfiguration(
+            provider_metadata=self.provider_metadata(registration_endpoint=registration_endpoint),
+            client_registration_info=ClientRegistrationInfo(
+                client_name='Test Client', post_logout_redirect_uris=['https://client.example.com/logout'],
+                redirect_uris=['https://client.example.com/swaggerui/oauth2-redirect.html'],
+                registration_token='initial_access_token'))
+
+        provider_config.register_client(Client(CLIENT_AUTHN_METHOD), 'https://client.example.com/redirect')
+
+        assert provider_config._client_metadata['client_id'] == 'client1'
+        assert provider_config._client_metadata['client_secret'] == 'secret1'
+        assert provider_config._client_metadata['client_name'] == 'Test Client'
+        assert provider_config._client_metadata['registration_client_uri'] == 'https://op.example.com/register/client1'
+        assert provider_config._client_metadata['registration_access_token'] == 'registration_access_token1'
+        assert provider_config._client_metadata['redirect_uris'] == [
+            'https://client.example.com/redirect', 'https://client.example.com/logout',
+            'https://client.example.com/swaggerui/oauth2-redirect.html']
+        assert provider_config._client_metadata[
+                   'post_logout_redirect_uris'] == ['https://client.example.com/logout']
+
+    def test_register_client_should_raise_value_error_if_type_redirect_uris_not_list(self):
```

Comment:
As mentioned above, I don't think this behavior belongs in `ProviderConfiguration` 🤔